### PR TITLE
disable cache creation and cleanup unused files

### PR DESCRIPTION
**Pull request by bluefish6**
Aims to address: https://github.com/pivotal/LicenseFinder/issues/685

Uncompressed image size reduced from 6.2 GB to 5.6 GB. No idea what will be the resulting compressed image size as reported by dockerhub though.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@bluefish6 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=686)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=686) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@bluefish6 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=pivotal/LicenseFinder&pullRequestId=686)!

File: Dockerfile [hunk 1]
```diff
@@ -51,8 +53,8 @@ RUN apt-get install -y python rebar
 
 # install and update python-pip
 RUN apt-get install -y python-pip python3-pip && \
-    pip2 install --upgrade pip==$PIP_INSTALL_VERSION  && \
-    pip3 install --upgrade pip==$PIP3_INSTALL_VERSION
+    pip2 install --no-cache --upgrade pip==$PIP_INSTALL_VERSION  && \
+    pip3 install --no-cache --upgrade pip==$PIP3_INSTALL_VERSION
```
Comment (xtreme-vikram-yadav) [inline Dockerfile]: @bluefish6 Did you mean `--no-cache-dir`? We were looking at available options for pip and we couldn't see `--no-cache`.

Comment (bluefish6) [reply Dockerfile]: You're right of course, I got the options wrong :)

```suggestion
    pip2 install --no-cache-dir --upgrade pip==$PIP_INSTALL_VERSION  && \
    pip3 install --no-cache-dir --upgrade pip==$PIP3_INSTALL_VERSION
```

File: Dockerfile [hunk 2]
```diff
@@ -123,11 +126,14 @@ RUN bash -lc "gem update --system && gem install bundler"
 
 # install conan
 RUN apt-get install -y python-dev && \
-	pip install --ignore-installed six --ignore-installed colorama --ignore-installed requests --ignore-installed chardet --ignore-installed urllib3 --upgrade setuptools && \
-    pip install -Iv conan==1.11.2
+	pip install --no-cache --ignore-installed six --ignore-installed colorama \
+	    --ignore-installed requests --ignore-installed chardet \
+	    --ignore-installed urllib3 \
+	    --upgrade setuptools && \
+    pip install --no-cache -Iv conan==1.11.2
```
Comment (xtreme-debbie-chen) [inline Dockerfile]: @bluefish6 Did you also mean `--no-cache-dir` here as well?

Comment (bluefish6) [reply Dockerfile]: Yeah, sorry, my bad.

```suggestion
    pip install --no-cache-dir -Iv conan==1.11.2
```

File: Dockerfile [hunk 3]
```diff
@@ -123,11 +126,14 @@ RUN bash -lc "gem update --system && gem install bundler"
 
 # install conan
 RUN apt-get install -y python-dev && \
-	pip install --ignore-installed six --ignore-installed colorama --ignore-installed requests --ignore-installed chardet --ignore-installed urllib3 --upgrade setuptools && \
-    pip install -Iv conan==1.11.2
+	pip install --no-cache --ignore-installed six --ignore-installed colorama \
```
Comment (pivotal-pmital) [inline Dockerfile]: oops @bluefish6, found another one. is this also supposed to be `--no-cache-dir`

Comment (bluefish6) [reply Dockerfile]: Of course you're right. Sorry, I should have checked the whole file when I got the first comment.
```suggestion
	pip install --no-cache-dir --ignore-installed six --ignore-installed colorama \
```

This seems to be the last one though ;)

File: Dockerfile [hunk 4]
```diff
@@ -156,7 +164,7 @@ RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 4F4EA0AAE5
 
 # install license_finder
 COPY . /LicenseFinder
-RUN bash -lc "cd /LicenseFinder && bundle install -j4 && rake install"
+RUN bash -lc "cd /LicenseFinder && bundle config set no-cache 'true' && bundle install -j4 && rake install"
```
Comment (xtreme-shane-lattanzio) [inline Dockerfile]: Sorry for another cache comment lol but will `bundle config --no-cache` work instead of `set no-cache 'true'`?

Comment (xtreme-shane-lattanzio) [reply Dockerfile]: @bluefish6 Bump :)

**Comment by xtreme-shane-lattanzio** [comment]
Hey @bluefish6 Im not sure if the notifications are sending in the comment so I just want to try to bump this one more time. Apologies if you are getting spammed with notifications.

**Pull request merged**